Lib/fontTools/ttLib/tables/_g_v_a_r.py

```diff
@@ -97,23 +137,17 @@ def decompile(self, data, ttFont):
 		offsets = self.decompileOffsets_(data[GVAR_HEADER_SIZE:], tableFormat=(self.flags & 1), glyphCount=self.glyphCount)
 		sharedCoords = tv.decompileSharedTuples(
 			axisTags, self.sharedTupleCount, data, self.offsetToSharedTuples)
-		self.variations = {}
+		self.variations = lazy_dict()
 		offsetToData = self.offsetToGlyphVariationData
 		glyf = ttFont['glyf']
+		def get_lambda(gvarData, numPointsInGlyph):
+			return lambda: decompileGlyph_(numPointsInGlyph, sharedCoords, axisTags, gvarData)
 		for i in range(self.glyphCount):
 			glyphName = glyphs[i]
 			glyph = glyf[glyphName]
 			numPointsInGlyph = self.getNumPoints_(glyph)
 			gvarData = data[offsetToData + offsets[i] : offsetToData + offsets[i + 1]]
-			try:
-				self.variations[glyphName] = decompileGlyph_(
-					numPointsInGlyph, sharedCoords, axisTags, gvarData)
-			except Exception:
-				log.error(
-					"Failed to decompile deltas for glyph '%s' (%d points)",
-					glyphName, numPointsInGlyph,
-				)
-				raise
+			self.variations[glyphName] = get_lambda(gvarData, numPointsInGlyph)
```

Comment:
Wouldn't this be prettier with a `functools.partial`?

Reply:
Thanks. Fixed.

Reply:
I was rather thinking that you could partial `decompileGlyph_` directly. I don't think `func` is really needed at all?

Reply:
Oh, you're right! Let me do that. Thanks.

Reply:
Done. So much nicer. Thanks.
https://github.com/fonttools/fonttools/pull/2738 is also ready :D.